nautobot/extras/context_managers.py

```diff
@@ -200,9 +218,12 @@ def web_request_context(
         request = RequestFactory().request(SERVER_NAME="web_request_context")
         request.user = user
     change_context = valid_contexts[context](request=request, context_detail=context_detail, change_id=change_id)
+    pre_object_data, pre_object_data_v2 = None, None
     try:
         with change_logging(change_context):
             yield request
+            change_context = change_context_state.get()
+            pre_object_data, pre_object_data_v2 = change_context.pre_object_data, change_context.pre_object_data_v2
```

Comment:
The `change_logging` context manager resets the change context upon exit, so we must capture the `before` object data here.

Reply:
https://github.com/nautobot/nautobot/pull/7213/commits/129cc29a5947ecc9bad3cd3a9737bfe03946ebd9